twenty-first/src/math/tip5.rs

```diff
@@ -673,8 +675,38 @@ impl Sponge for Tip5 {
     }
 }
 
+impl Hasher for Tip5 {
+    fn write(&mut self, bytes: &[u8]) {
+        // Convert input bytes into BFieldElements and absorb them in chunks of `RATE`
+        let mut bfield_elements = bytes
+            .chunks(8) // Assuming each `BFieldElement` corresponds to 8 bytes
+            .map(|chunk| {
+                let mut buffer = [0u8; 8];
+                buffer[..chunk.len()].copy_from_slice(chunk);
+                BFieldElement::from_raw_u64(u64::from_le_bytes(buffer))
+            });
+
+        // Absorb the data in chunks
+        while let Ok(chunk) = bfield_elements
+            .by_ref()
+            .take(Tip5::RATE)
+            .collect::<Vec<_>>()
+            .try_into()
+        {
+            self.absorb(chunk);
+        }
+    }
+
+    /// Produce a 64-bit hash value from the sponge's state
+    fn finish(&self) -> u64 {
+        // Extract the first part of the state as the hash value
+        self.state[0].raw_u64()
```

Comment:
Similar to above, I strongly suggest avoiding the `raw` methods.

```suggestion
        self.state[0].value()
```